### PR TITLE
Updates to the RAM-SCB documentation v0.1

### PR DESCRIPTION
**Pull request by Pheosics**
First update to the RAM-SCB documentation. Contains questions that need
answering in order to continue with the documentation updates

### CODE REVIEW

**Comment by Pheosics** [comment]
@drsteve Would you please look through the .tex files and look for comments marked `% QUESTION:`. Need some input on how to continue with the documentation

**Comment by Pheosics** [comment]
@drsteve We should also consider having two separate documents, a user's manual (like this one) and a developers manual. Could also just expand on your WORKFLOW document, perhaps turn it into a CONTRIBUTING document with the information from the WORKFLOW document plus our expectations for code submissions, for example, if we have any code structure requirements (e.g. no GOTO statements), require tests be written, require new PARAMS added to the PARAM.XML, etc...

File: doc/Tex/RAM_SCB_install.tex [hunk 1]
```diff
@@ -1,180 +1,72 @@
...
-message and continue to the next step.
-
-Install Pspline as follows (changing the value of {\tt PREFIX} to select 
-installation directory):
-\begin{verbatim}
-make install PREFIX=~/psline_portland
-\end{verbatim}
-On machines where you will use different compilers, it is helpful to add the 
-compiler name to the name of the installation directory.
+% QUESTION: All of the information in this section is out of date. Even the NETCDF installation instructions are incorrect given that they have split it into a C version and a Fortran version. This is still easy enough to describe and walk through, but the installation of GSL from source isn't as easy. Do we really want to have this section?
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:10]: I'm not convinced that our documentation needs to have install instructions for dependencies. My preference would be to say what the dependencies are, point to where they are (e.g., unidata for NetCDF), and note that everything needs to be compiled with the same compiler. If there are gotchas - like NetCDF-C and NetCDF-Fortran having different version numbers - then we should probably note that.

Where we will want more detailed instructions is in making sure that RAM-SCB finds the libraries.

File: doc/Tex/RAM_SCB_install.tex [hunk 2]
```diff
@@ -1,180 +1,72 @@
...
 
 \subsection{Libraries From Modules}
-Well maintained clusters often use the {\tt module} interface for loading
-and unloading software packages.  Be sure to explore the available packages
-on a new system before going through the arduous task of installing from source!
+Well maintained clusters often use the {\tt module} interface for loading and unloading software packages.  Be sure to explore the available packages on a new system before going through the arduous task of installing from source!
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:13]: Most places with a `module` system also have support... It's probably spoon-feeding, but maybe say that before installing from source it's a good idea to see if the cluster support can build modules for you.

File: doc/Tex/RAM_SCB_install.tex [hunk 3]
```diff
@@ -1,180 +1,72 @@
...
 \begin{verbatim}
 module load package_name
 module unload package_name
 module list
 \end{verbatim}
 
-As an example, here is what you would type on NASA's Pleiades cluster to get
-started with RAM-SCB:
+As an example, here is what you would type on NASA's Pleiades cluster to get started with RAM-SCB:
+% QUESTION: ^Is this even still true? Seems silly to have to keep up with changes on Pleiades in our documentation
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:28]: No idea. I haven't used Pleiades and I don't know what the range of modules there are. Having an example is useful though, so maybe we just switch to the grizzly modules and say it's an illustrative example. Probably also good to mention `module purge` to ensure that everything is unloaded...

File: doc/Tex/RAM_SCB_install.tex [hunk 4]
```diff
@@ -1,180 +1,72 @@
...
+module load gsl/2.4
 module load python/2.6.1
 \end{verbatim}
 
-Intel Fortran, MPI, NetCDF and NCAR Graphics are now loaded and ready to use.
-Place such commands your shell configuration 
-script ({\tt \verb*l~l/.cshrc} or {\tt \verb*l~l/.bashrc} depending on your 
-shell) to load modules upon login.  Do not forget to point RAM-SCB to these
-libraries as listed below.
+Intel Fortran, MPI, NetCDF and GSL are now loaded and ready to use. Place such commands your shell configuration script ({\tt \verb*l~l/.cshrc} or {\tt \verb*l~l/.bashrc} depending on your shell) to load modules upon login. Do not forget to point RAM-SCB to these libraries as listed below.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:38]: This approach is okay, and I'm not going to recommend folks *don't* do it (especially if you're running on a local server), but most job schedulers on HPC launch the jobs in a new environment, so modules loaded from the `rc` file may not propagate to your job.
Note here that the module may need to be loaded in a job script on systems managed by a job scheduler? (e.g., slurm, moab, pbs pro)

File: doc/Tex/RAM_SCB_install.tex [hunk 5]
```diff
@@ -1,180 +1,72 @@
...
 
-Finally, if you set neither the environment variables or the Config.pl 
-switches, the Fortran compiler will look for the libaries in the default 
-directories.  This may be useful to test if the libraries are already 
-properly installed by the system administrator.  
+Finally, if you set neither the environment variables or the Config.pl switches, the Fortran compiler will look for the libaries in the default directories. This may be useful to test if the libraries are already properly installed by the system administrator.  
 
-\emph{If you change the library locations by any method, 
-you must reinstall the code.} 
+\emph{If you change the library locations by any method, you must reinstall the code.}
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:53]: For those with less experience, it might be nice to point out that the library locations are found at runtime. So at runtime they need to be in the same place - which means that if you compile with a certain set of modules loaded, the configuration must be the same at runtime.

File: doc/Tex/RAM_SCB_install.tex [hunk 6]
```diff
@@ -187,41 +79,27 @@ \section{Installation, Configuration, and Compiling \label{subchap:install}}
...
-\item{Use {\tt -compiler} to select the Fortran 90 compiler.  
-  Common choices include pgf90, ifort, gfortran, and f95(for NAG).}
-\item{The {\tt -mpi} flag allows the user to pick which version of MPI to use.
-  Choose from mpich, mpich2, Altix, and openmpi.  Alternatively, the 
-  {\tt -nompi} flag may be set with no value.  This option is fragile.}
-\item{Set {\tt -ncdf} to the path of the NetCDF library installation.  
-  If used, this flag overrides the environment variable NETCDFDIR.}
-\item{Set {\tt -pspline} to the path of the Pspline library installation.  
-  If used, this flag overrides the environment variable PSPLINEDIR.}
+\item{Use {\tt -compiler} to select the Fortran 90 compiler. Common choices include pgf90, ifort, gfortran, and f95(for NAG).}
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:84]: Should these be listed in order of our preference? That is, we test on `gfortran` so list it first... then probably `ifort`, then anything else.

File: doc/Tex/RAM_SCB_install.tex [hunk 7]
```diff
@@ -187,41 +79,27 @@ \section{Installation, Configuration, and Compiling \label{subchap:install}}
...
+\item{The {\tt -mpi} flag allows the user to pick which version of MPI to use. Choose from mpich, mpich2, Altix, and openmpi. Alternatively, the {\tt -nompi} flag may be set with no value. This option is fragile.}
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_install.tex:85]: Same as previous comment here. I'd list in order that we're interested in using or supporting. Probably `openmpi`, then `mpich/mpich2`, then `altix`

File: doc/Tex/RAM_SCB_intro.tex [hunk 8]
```diff
@@ -1,56 +1,17 @@
...
+
+RAM-SCB has grown from a research-grade code with limited options and static magnetic field (RAM) to a rich, highly configurable research and operations tool with a multitude of new physics and output products.  This manual providesa guide to users who want to learn how to install, configure, and execute RAM-SCB simulations.  While the code is designed to make these steps asstraight-forward as possible, it is strongly recommended that users review the publications listed in the Bibliography to ensure a thorough understanding of the physics included in the model.  Additionally, all users are asked to review the terms of use.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_intro.tex:3]: Couple of spaces missing in this block of text

File: doc/Tex/RAM_SCB_intro.tex [hunk 9]
```diff
@@ -1,56 +1,17 @@
...
 
-Use of the RAM-SCB software implies agreement with the terms herein.
-RAM-SCB is open source software that has been developed at Los Alamos National Laboratory (LANL). The code is based on physics and
-numerical methods detailed in the following publications and references therein:
+Use of the RAM-SCB software implies agreement with the terms herein. RAM-SCB is open source software that has been developed at Los Alamos National Laboratory (LANL). The code is based on physics and numerical methods detailed in the following publications and references therein:
 
-1. Jordanova, V. K. et al. (2006), Kinetic simulations of ring current 
-evolution during the Geospace Environment Modeling challenge events, 
-J. Geophys, Res., 111, A11S10, doi:10.1029/2006JA011644. \\
+1. Jordanova, V. K. et al. (2006), Kinetic simulations of ring current evolution during the Geospace Environment Modeling challenge events, J. Geophys, Res., 111, A11S10, doi:10.1029/2006JA011644. \\
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_intro.tex:12]: These need updating. Recent Welling publication, plus the Engel paper. As you know, a lot has changed and those papers aren't really up-to-date descriptions any more.

File: doc/Tex/RAM_SCB_intro.tex [hunk 10]
```diff
@@ -90,9 +43,9 @@ \section{TERMS OF USE \& DISTRIBUTION POLICY}
     \hline
     Welling et al. [2011] & First full description of one-way coupling with SWMF \\
     \hline
-    Welling et al. [2015] & Description of two-way coupling of RAM-SCB
-  with SWMF \\
+    Welling et al. [2015] & Description of two-way coupling of RAM-SCB with SWMF \\
     \hline
+    Engel et al. [2019] & Improved simulations of the inner magnetosphere during high geomagnetic activity with the RAM‐SCB model \\
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_intro.tex:48]: As above, I'd think about revising the list of what counts as "physics" versus "development milestones".

File: doc/Tex/RAM_SCB_quickstart.tex [hunk 11]
```diff
@@ -1,38 +1,23 @@
...
 
 \section{Installation}
 
-Installation and compilation of RAM-SCB requires a Fortran compiler, 
-Perl interpreter, an MPI library compiled with the preferred Fortran compiler, 
-and several external libraries which have 
-additional 
-requirements (most notably a C compiler, typically standard on Unix-like 
-systems.)  The required libraries are:
+Installation and compilation of RAM-SCB requires a Fortran compiler, Perl interpreter, an MPI library compiled with the preferred Fortran compiler, and several external libraries which have additional requirements (most notably a C compiler, typically standard on Unix-like systems.)  The required libraries are:
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_quickstart.tex:5]: Compiler modern enough to have OpenMP preferred...

File: doc/Tex/RAM_SCB_run.tex [hunk 12]
```diff
@@ -1,95 +1,48 @@
...
-let's pretend we're using NASA's Pleiades system and doing a set of validation
-simulations.  On this system, you typically install code on your home 
-directory, but must run from designated directories called ``nobackupXX'' 
-(where XX is the number of the directory and nobackup tells you why you 
-didn't install your code there.)  From the install directory, make and move
-your first run directory:
+{\tt make} will unpack some default inputs and organize a new directory aptly named {\tt run}. Note that if a run directory exists that is called ``run'', RAM-SCB will not over write it and {\tt make} will complain. 
 
+The first thing you should do is rename your run directory and move it to a place that makes sense given the constraints of your system. For example, let's pretend we're using NASA's Pleiades system and doing a set of validation simulations. On this system, you typically install code on your home directory, but must run from designated directories called ``nobackupXX'' (where XX is the number of the directory and nobackup tells you why you didn't install your code there.) From the install directory, make and move your first run directory:
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_run.tex:9]: Again with the Pleiades... This is a generic "on HPC you should run on scratch" statement, that we should probably make generically.

File: doc/Tex/RAM_SCB_run.tex [hunk 13]
```diff
@@ -1,95 +1,48 @@
...
-to tell RAM-SCB what and how it should simulate.  The {\tt PARAM} interface,
-identical to the one used by the SWMF, is simple.  Any line of text in the
-file not preceeded by a \# symbol is ignored, so comment your file to your
-heart's content.    Any text preceeded by \#, however, is a interpreted as
-a {\tt PARAM} command.  A generic command takes the form,
+Figure \ref{fig:rundir} shows the default run directory organization. The IM directory is where most input and output occurs; subdirectories shaded green are soft-linked to the top run level for convenience. These directories are self explanatory outside of {\tt output\_swmf}, which is for runs that use SWMF output in stand-alone mode. Enough inputs are given in the input directories for a sample simulation right away. The red box in Figure \ref{fig:rundir} denotes the linked executable. The final files, denoted as yellow boxes, are the final input files. {\tt w2k.dat} is the Weimer 2000 empirical model data file, {\tt omni.txt} is solar wind input file from the OMNI database, and {\tt RamIndices.txt} is a list of geomagnetic indices. The necessity of these three files is discussed below.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_run.tex:33]: Are the required formats for, e.g., the OMNI file documented here? What is in which column, whether it requires a specific format statement, delimiter, etc.

File: doc/Tex/RAM_SCB_run.tex [hunk 14]
```diff
@@ -139,115 +82,48 @@
...
+
+Throughout a simulation, restart files are being written to the {\tt restartOUT} folder of the run directory. The files are written at a set frequency and at the end of a successful run (both of these options are configuarble through the {\tt PARAM} interface.) Only the most recent restart is saved, users can periodically pull these files out of {\tt restartOUT} if they prefer a back log of restarts. These files contain the full distribution functions as well as the start time, current time, and current iteration at the point that the restart was written.
+
+To restart a run, first move restart files from {\tt restartOUT} to {\tt restartIN}. Then, either create a new {\tt PARAM.in} file or edit the existing one such that no \verb*l#STARTTIMEl command is present (remember, the restart file knows this information already), the run time exceeds the current time listed in the restart file, and the command \verb*l#RESTARTl is included. This tells RAM-SCB to restart the simulation rather than start anew. Restarts are carefully implemented to preserve files in the output directories and continue the simulation as if there was no interruption.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_run.tex:100]: Is this still required? I thought we'd changed this at some point to have the latest restart linked/copied over to the restartIN directory automatically... I may well be misremembering this.

File: doc/Tex/RAM_SCB_run.tex [hunk 15]
```diff
@@ -139,115 +82,48 @@
...
+ \begin{tabular}{l l l l l}
+ \hline\hline
+  Type & Extension & Format & Default? & Contents\\
+ \hline
+ Log file & \verb*q*.logq & ASCII & Yes & Dst, integrated values\\
+ Pressure files & \verb*l*.inl & ASCII & Yes & $\perp$ and $\parallel$ partial pressures\\
+ Trapped files & \verb*l*.tl & ASCII & No & Averaged trapped fluxes\\
+ E-Field files & \verb*l*.inl & ASCII & No & Equatorial electric potential\\
+ Boundary files & \verb*l*.datl & ASCII & No & Fluxes at the outer boundary\\
+ Full flux files & \verb*l*.ncl & NetCDF & Yes & Full equatorial flux information\\
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_run.tex:119]: If these are equatorial only, why are they called "full"? Okay, I know the answer to that, but it's still a little misleading... Is there any reason we can't change this to be "Equatorial Flux Files" and "3D Flux Files"?

File: doc/Tex/RAM_SCB_scripts.tex [hunk 16]
```diff
@@ -1,73 +1,39 @@
...
 
-Common options include {\tt -a=VALUE} and {\tt -r=VALUE}, which allow the user
-to ignore absolute and relative differences less than {\tt VALUE} and {\tt -t}
-which turns off the comparison of text.
+Common options include {\tt -a=VALUE} and {\tt -r=VALUE}, which allow the user to ignore absolute and relative differences less than {\tt VALUE} and {\tt -t} which turns off the comparison of text.
 
 
 %\section{GetOmni.py (RAM-SCB Script) \label{subchap:getomni}}
 %Script forthcoming.
+% QUESTION: We sort of have this now, should we make a section for it?
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_scripts.tex:27]: Yes we should add a section, but we don't have this *in* in the repo, do we? We should make sure the script is in the repo, then make sure the section documenting use is present and correct.

File: doc/Tex/RAM_SCB_scripts.tex [hunk 17]
```diff
@@ -1,73 +1,39 @@
...
 \end{verbatim}
-\noindent will print out the script's help text.  This info is typically 
-more complete and up-to-date than the info listed here.
+\noindent will print out the script's help text.  This info is typically more complete and up-to-date than the info listed here.
 
-While all RAM-SCB scripts are listed here, only a handful of SWMF scripts are
-described.  Be sure to explore the {\tt Share/Scripts} directory before you
-constructing your own solutions to common problems (e.g. endian conversions, 
-PARAM-checking, etc.) 
+While all RAM-SCB scripts are listed here, only a handful of SWMF scripts are described. Be sure to explore the {\tt Share/Scripts} directory before you constructing your own solutions to common problems (e.g. endian conversions, PARAM-checking, etc.)
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_scripts.tex:9]: This section also needs to have the `updateRAMindices` script. Not sure we need to document the "rt"-oriented scripts, although we should check whether everything is Python3 compatible.

File: doc/Tex/RAM_SCB_swmf.tex [hunk 18]
```diff
@@ -1,22 +1,13 @@
-RAM-SCB may be used as a component to the Space Weather Modeling Framework.
-Under this mode, RAM-SCB receives initial and boundary conditions from the
-Framework's other components and returns plasma properties to create a two-way
-coupled system.
+RAM-SCB may be used as a component to the Space Weather Modeling Framework. Under this mode, RAM-SCB receives initial and boundary conditions from the Framework's other components and returns plasma properties to create a two-way coupled system.
 
 %INSERT TABLE OF COUPLING DESCRIPTIONS.
 
 \section{Installing RAM-SCB as a SWMF Component}
+% QUESTION: Should this entire section just be rewritten? SWMF no longer uses CVS, we have changed the coupling on our end a bit. And SWMF has changed the coupling on their end a lot. Honestly, it may be of more use to have an entirely seperate document for the SWMF coupling as it has its own set of installation instructions and its own set of difficulties.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_swmf.tex:6]: Yes, should be rewritten. I'd start with the assumption of having either a tarball from the CSEM webpage or a checked-out copy of the repo. A new chapter is probably a good place for it, and minimal but correct is better than verbose and incorrect/outdated.

File: doc/Tex/RAM_SCB_testing.tex [hunk 19]
```diff
@@ -1,77 +1,57 @@
-RAM-SCB comes packaged with a test suite to determine if the current 
-installation operates correctly and produces the expected results.  
-Different tests evaluate different code capabilities and options.  
-Testing is an especially powerful development tool for evaluating the impact, 
-intentional or not, of changes to the source code.  A summary of the tests
-and associated commands can be generated by using {\tt make test\_help}.
+RAM-SCB comes packaged with a test suite to determine if the current installation operates correctly and produces the expected results. Different tests evaluate different code capabilities and options. Testing is an especially powerful development tool for evaluating the impact, intentional or not, of changes to the source code. A summary of the tests and associated commands can be generated by using {\tt make test\_help}.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_testing.tex:1]: We should tell the user to check current test status by visiting the github repo.

File: doc/Tex/RAM_SCB_testing.tex [hunk 20]
```diff
@@ -1,77 +1,57 @@
...
 
-If, at any point, the test fails, the details will be recorded in 
-\textit{nametest}.diff in the installation directory.  If the test is 
-successful, this file will be created but be empty.  Tests that fail at the 
-comparitive stage will list all of the differences.  When this happens, 
-additional tools to examine the file differences are recommended.  An 
-excellent visualization of file differences can be rendered with 
-\verb*ltkdiffl, an open source tool that is already installed on many 
-Linux machines.
+If, at any point, the test fails, the details will be recorded in \textit{nametest}.diff in the installation directory. If the test is successful, this file will be created but be empty. Tests that fail at the comparitive stage will list all of the differences. When this happens, additional tools to examine the file differences are recommended. An excellent visualization of file differences can be rendered with \verb*ltkdiffl, an open source tool that is already installed on many Linux machines.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_testing.tex:25]: `vimdiff` or `vim -d` (they're basically equivalent) is a good command line option. Similarly, `gvimdiff` is much lighter weight than tkdiff but is also gui based.

File: doc/Tex/RAM_SCB_testing.tex [hunk 21]
```diff
@@ -1,77 +1,57 @@
...
 
 \subsection{Test 1}
 
 \textbf{Command:} \verb*lmakel \verb*ltest1l
 
-This is the most basic test for RAM-SCB.  It runs RAM-SCB for 300 seconds using
-Volland-Stern electric field, constant dipole magnetic field (no SCB 
-calculation), and LANL fluxes.  \textbf{SHIELDS-RC} users should rely on this
-test.
+This is the most basic test for RAM-SCB. It runs RAM-SCB for 900 seconds using Volland-Stern electric field, constant dipole magnetic field (no SCB calculation), and LANL fluxes. \textbf{SHIELDS-RC} users should rely on this test.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_testing.tex:33]: Do we want discussion of SHIELDS-RC here? Is that mode still a directive in the PARAM?
If we've nixed it then we need to clean up discussion of SHIELDS-RC...

File: doc/Tex/RAM_SCB_testing.tex [hunk 22]
```diff
@@ -1,77 +1,57 @@
...
-Test 3 activates SCB, uses the more complicated Tsyganenko 89 boundary 
-conditions and Weimer 2000 empirical electric field.  This is the base test
-for SCB functionality and ionosphere-to-equator electric potential mapping.
+Test 3 activates SCB, uses the more complicated Tsyganenko 89 boundary conditions and Weimer 2000 empirical electric field. This is the base test for SCB functionality and ionosphere-to-equator electric potential mapping.
 
+\subsection{Test 4}
+
+\textbf{Command:} \verb*lmakel \verb*ltest4l
+
+Test 4 repeats the previous test, but stops half way through the simulation, writes restart files, and restarts the run. This exercises the code's ability to seamlessly restart a simulation at any arbitrary point.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_testing.tex:51]: "... like test2, this exercises the code's ability to ..."

File: doc/Tex/RAM_SCB_viz.tex [hunk 23]
```diff
@@ -1,12 +1,5 @@
-RAM-SCB output can be visualized in a number of different ways depending on
-the user's tastes and preferences.  However, a standard library for opening,
-manipulating, and plotting exists as a sub-module of Spacepy.  Users may
-find this freely available Python library at
-\href{http://spacepy.lanl.gov/}{spacepy.lanl.gov}.  Alternatively,
-development versions can be obtained via the project's
-\href{http://sourceforge.net/projects/spacepy/}{Source Forge} page.  The
-RAM-SCB module is nested under {\tt spacepy.pybats.ram}.
+RAM-SCB output can be visualized in a number of different ways depending on the user's tastes and preferences.  However, a standard library for opening, manipulating, and plotting exists as a sub-module of Spacepy.  Users may find this freely available Python library at \href{http://spacepy.lanl.gov/}{spacepy.lanl.gov}.  Alternatively, development versions can be obtained via the project's \href{http://sourceforge.net/projects/spacepy/}{Source Forge} page.  The RAM-SCB module is nested under {\tt spacepy.pybats.ram}.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_viz.tex:1]: spacepy.lanl.gov has been dead for a while, as has spacepy's sourceforge page.

These need updating to https://spacepy.github.io/ (docs), and https://github.com/spacepy/spacepy (repo).

File: doc/Tex/RAM_SCB_viz.tex [hunk 24]
```diff
@@ -1,12 +1,5 @@
...
+% QUESTION: Steve, do we need to update the spacepy location (moved to git didn't it?)? Also, should we expand on this section to show actual examples of the visualization?
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_viz.tex:2]: As above, yes to the updated location.

I'd recommend doing the expansion with some simple examples, but maybe as a separate PR once this one is done.

File: doc/Tex/RAM_SCB_viz.tex [hunk 25]
```diff
@@ -1,12 +1,5 @@
...
 
-Stand-alone code snippets for both Python and Matlab can be found in the
-{viz} directories.  These routines are largely out-of-date and not maintained.
-We recommend caution when using them.
+Stand-alone code snippets for both Python and Matlab can be found in the {viz} directories.  These routines are largely out-of-date and not maintained. We recommend caution when using them.
+% QUESTION: ^ This seems like a silly comment, maybe we should add new stand-alone code snippets that actually work, or possibly code snippets that use spacepy as an example? I have both.
```
Comment (drsteve) [inline doc/Tex/RAM_SCB_viz.tex:5]: The `viz` directory that's referring to was removed. 

The viz directory that exists now is `Scripts/viz` and that has tools for visualizing the domain (from restart files) using VTK and Paraview.

File: doc/Tex/RAM_SCB_install.tex (hunk 1 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_install.tex:10]: That sounds good to me, especially since we don't have anything weird like PSpline now

File: doc/Tex/RAM_SCB_install.tex (hunk 2 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_install.tex:13]: Hand holding and spoon feeding aren't necessarily bad things. If it will get more people to use the code.

File: doc/Tex/RAM_SCB_install.tex (hunk 4 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_install.tex:38]: Thats a good point, I'll add the clarification

File: doc/Tex/RAM_SCB_install.tex (hunk 5 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_install.tex:53]: Good point

File: doc/Tex/RAM_SCB_install.tex (hunk 6 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_install.tex:84]: I'd say we just list gfortran, but I understand that other people like other compilers so listing them in preference order is probably best. We should also note that a c compiler is required since we use GSL. Do ifort and the others have c compatible compilers?

File: doc/Tex/RAM_SCB_install.tex (hunk 7 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_install.tex:85]: Done

File: doc/Tex/RAM_SCB_quickstart.tex (hunk 11 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_quickstart.tex:5]: Yes

File: doc/Tex/RAM_SCB_run.tex (hunk 12 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_run.tex:9]: K

File: doc/Tex/RAM_SCB_run.tex (hunk 13 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_run.tex:33]: Yes, I think we need a new section for input formats. No information is currently available for that

File: doc/Tex/RAM_SCB_run.tex (hunk 14 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_run.tex:100]: That was the plan I believe, it doesn't currently happen though. But it should be a simple enough fix

File: doc/Tex/RAM_SCB_run.tex (hunk 15 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_run.tex:119]: No reason I can think of. Actually, most of those output files don't even exist any more. They were either changed in format, or were merged with another file. There have also been several additional file outputs added. So that whole section needs to get redone

File: doc/Tex/RAM_SCB_scripts.tex (hunk 16 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_scripts.tex:27]: I don't believe the script is currently in the repo, good point that it probably should be if we are going to document it

File: doc/Tex/RAM_SCB_scripts.tex (hunk 17 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_scripts.tex:9]: Good point

File: doc/Tex/RAM_SCB_testing.tex (hunk 21 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_testing.tex:33]: It is still there, and still sets the appropriate flags

File: doc/Tex/RAM_SCB_viz.tex (hunk 23 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_viz.tex:1]: K

File: doc/Tex/RAM_SCB_install.tex (hunk 6 above)
Comment (drsteve) [reply doc/Tex/RAM_SCB_install.tex:84]: Intel has `icc` and `ifort`. Portland has `pgcc` (among different variants) and its various fortran compilers.
Not sure about NAG. I can't find out anything about a C compiler (just NAG numerical libraries and the Fortran compiler).

File: doc/Tex/RAM_SCB_viz.tex (hunk 24 above)
Comment (Pheosics) [reply doc/Tex/RAM_SCB_viz.tex:2]: K

File: doc/Tex/RAM_SCB_intro.tex (hunk 8 above)
Comment (drsteve) [reply doc/Tex/RAM_SCB_intro.tex:3]: Just noticed "to the shell created by field lines passing through the SM equatorial plane at radial distances which overlap the RAM domain." The SCB domain isn't defined that way since the grids were decoupled... right?

File: doc/Tex/RAM_SCB_run.tex (hunk 14 above)
Comment (drsteve) [reply doc/Tex/RAM_SCB_run.tex:100]: Fair enough. Let's keep this for now and then when the update gets made to the code the docs can be updated.

**Comment by drsteve** [comment]
Hi @Pheosics - it looks like you maybe you merged master into your documentation branch to update (instead of rebasing), so now there's a bunch of duplicated commits. It's probably a good idea for one of us to clean this up, as I think in the current state it'll have issues with a rebase merge into the repo... if you want me to do it, LMK when you think you'll next be working on this and I'll try to get it done by then so you can just and work from the updated branch.

**Comment by drsteve** [comment]
Since this is documentation and not code, I'm happy to take this one over and then merge after someone/anyone has given it a quick once-over. I've got a little time to chip away at it, so unless I see otherwise here I'm taking this PR over...

**Comment by Pheosics** [comment]
Sounds good to me. I don't have any uncommited changes, so everything that is in the pull request should be up to date. If there are any specific pieces you think would make sense for me to write/document you can assign them to me if you want and I will try my best to get a quick turn around (should be easier to find time to work on one or two small pieces as opposed to the whole thing)

File: doc/Tex/RAM_SCB_run.tex (hunk 13 above)
Comment (drsteve) [reply doc/Tex/RAM_SCB_run.tex:37]: I'm going to leave this as future work so we can get a first cut committed. Marking as resolved.

File: doc/Tex/RAM_SCB_swmf.tex (hunk 18 above)
Comment (drsteve) [reply doc/Tex/RAM_SCB_swmf.tex:12]: I've left old content and our notes as comments, with a placeholder directing the user to seek assistance. Marking as resolved for now, this can be fleshed out in the near future - I'm mostly looking to get something not-out-of-date committed as a new baseline.

**Comment by drsteve** [comment]
Okay... I've made a few updates and have resolved anything outstanding in our previous conversations that had a chance of being updated right now. If we have a baseline set of usable docs in place we can just update as we go.

The PDF in the repo reflects all of the changes, so if someone (@Pheosics ?) can leaf through and say whether there's anything egregious in there... If you don't have time, that's fine, just drop a line here and I can do _the bad thing_ and merge it w/o sign-off. It's only docs so no code will get broken. Current docs are also wrong/obsolete, so this has to be an improvement!

**Comment by Pheosics** [comment]
@drsteve Looks good, the only thing that caught my eye was the USEFLC command in the PARAM.xml. I believe there should be a empty line between the variables and the description text string. Otherwise the text string is treated as a variable and doesn't wrap.
